### PR TITLE
typescript: index 'extends' clauses in interfaces

### PR DESCRIPTION
**Pull request by evmar**
The argument to an 'extends' clause is a type, not a value.
This matters when Interface1 extends Interface2, as the latter
is *only* a type and not a value.

### CODE REVIEW

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -392,6 +392,22 @@ class Vistor {
...
   }
 
+  /**
+   * visitHeritage visits the X found in an 'extends X' or 'implements X'.
+   *
+   * These are subtle in an interesting way.  When you have
+   *   interface X extends Y {}
+   * that is referring to the *type* Y (because interfaces are types, not
+   * values).  But it's also legal to write
+   *   class X extends (class Z { ... }) {}
```
Comment (zrlk) [inline kythe/typescript/indexer.ts]: Could you add a test for this case?

**Comment by schroederc** [comment]
Any progress on this?

**Comment by evmar** [comment]
In https://github.com/google/kythe/pull/2838 I add a test with some class expressions, and in this change (based on that) I added some more tests.

File: kythe/typescript/indexer.ts (hunk 1 above)
Comment (evmar) [reply kythe/typescript/indexer.ts]: Done

**Pull request merged**